#libby #build

The `com.ibm.ws.microprofile.graphql_fat_tck` bucket failed with javacores. It looks like the child build hit an `OutOfMemoryError`

```
[19-Jul-2022 02:24:18:188 PDT] 00000021 JVMDUMP039I Processing dump event "systhrow", detail "java/lang/OutOfMemoryError" at 2022/07/19 02:24:18 - please wait.
[19-Jul-2022 02:24:18:193 PDT] 00000021 JVMDUMP032I JVM requested System dump using '/home/jazz_build/Build/jbe/build/dev/core.20220719.022418.22618.0001.dmp' in response to an event
[19-Jul-2022 02:24:37:793 PDT] 00000021 JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/share/apport/apport %p %s %c %d %P %E" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.22753.
[19-Jul-2022 02:24:37:806 PDT] 00000021 
[19-Jul-2022 02:24:43:105 PDT] 00000021 JVMDUMP012E Error in System dump: The core file created by child process with pid = 22753 was not found. Expected to find core file with name "/home/jazz_build/Build/jbe/build/dev/core"
[19-Jul-2022 02:24:43:106 PDT] 00000021 JVMDUMP032I JVM requested Heap dump using '/home/jazz_build/Build/jbe/build/dev/heapdump.20220719.022418.22618.0002.phd' in response to an event
[19-Jul-2022 02:24:49:018 PDT] 00000021 JVMDUMP010I Heap dump written to /home/jazz_build/Build/jbe/build/dev/heapdump.20220719.022418.22618.0002.phd
[19-Jul-2022 02:24:49:019 PDT] 00000021 JVMDUMP032I JVM requested Java dump using '/home/jazz_build/Build/jbe/build/dev/javacore.20220719.022418.22618.0003.txt' in response to an event
[19-Jul-2022 02:24:49:119 PDT] 00000021 JVMDUMP010I Java dump written to /home/jazz_build/Build/jbe/build/dev/javacore.20220719.022418.22618.0003.txt
[19-Jul-2022 02:24:49:121 PDT] 00000021 JVMDUMP032I JVM requested Snap dump using '/home/jazz_build/Build/jbe/build/dev/Snap.20220719.022418.22618.0004.trc' in response to an event
[19-Jul-2022 02:24:49:122 PDT] 00000021 JVMDUMP010I Snap dump written to /home/jazz_build/Build/jbe/build/dev/Snap.20220719.022418.22618.0004.trc
[19-Jul-2022 02:24:49:122 PDT] 00000021 JVMDUMP013I Processed dump event "systhrow", detail "java/lang/OutOfMemoryError".
[19-Jul-2022 02:24:49:123 PDT] 00000021 JVMDUMP039I Processing dump event "systhrow", detail "java/lang/OutOfMemoryError" at 2022/07/19 02:24:49 - please wait.
[19-Jul-2022 02:24:49:124 PDT] 00000021 JVMDUMP032I JVM requested Heap dump using '/home/jazz_build/Build/jbe/build/dev/heapdump.20220719.022449.22618.0005.phd' in response to an event
[19-Jul-2022 02:24:55:033 PDT] 00000021 JVMDUMP010I Heap dump written to /home/jazz_build/Build/jbe/build/dev/heapdump.20220719.022449.22618.0005.phd
[19-Jul-2022 02:24:55:033 PDT] 00000021 Target 'getImage' failed with message 'java.lang.OutOfMemoryError: Java heap space'.
[19-Jul-2022 02:24:55:034 PDT] 00000021 Cannot execute 'invokeAsyncTarget' - 'getImage' failed or was not executed.
[19-Jul-2022 02:24:55:034 PDT] 00000021 Cannot execute 'launchFatTests' - 'invokeAsyncTarget' failed or was not executed.
[19-Jul-2022 02:24:55:035 PDT] 00000021 JVMDUMP032I JVM requested Java dump using '/home/jazz_build/Build/jbe/build/dev/javacore.20220719.022449.22618.0006.txt' in response to an event
[19-Jul-2022 02:24:55:137 PDT] 00000021 JVMDUMP010I Java dump written to /home/jazz_build/Build/jbe/build/dev/javacore.20220719.022449.22618.0006.txt
[19-Jul-2022 02:24:55:138 PDT] 00000021 JVMDUMP032I JVM requested Snap dump using '/home/jazz_build/Build/jbe/build/dev/Snap.20220719.022449.22618.0007.trc' in response to an event
[19-Jul-2022 02:24:55:138 PDT] 00000021 JVMDUMP010I Snap dump written to /home/jazz_build/Build/jbe/build/dev/Snap.20220719.022449.22618.0007.trc
[19-Jul-2022 02:24:55:138 PDT] 00000021 JVMDUMP013I Processed dump event "systhrow", detail "java/lang/OutOfMemoryError".
[19-Jul-2022 02:24:55:139 PDT] 00000021 
[19-Jul-2022 02:24:55:139 PDT] 00000021 BUILD FAILED
[19-Jul-2022 02:24:55:139 PDT] 00000021 /home/jazz_build/Build/jbe/build/dev/ant_build/build-liberty-async.xml:36: java.lang.OutOfMemoryError: Java heap space
[19-Jul-2022 02:24:55:140 PDT] 00000021 	at java.base/java.lang.StringUTF16.newBytesFor(StringUTF16.java:52)
[19-Jul-2022 02:24:55:140 PDT] 00000021 	at java.base/java.lang.String.replace(String.java:2482)
[19-Jul-2022 02:24:55:141 PDT] 00000021 	at org.apache.tools.ant.util.FileUtils.isAbsolutePath(FileUtils.java:660)
[19-Jul-2022 02:24:55:141 PDT] 00000021 	at org.apache.tools.ant.util.FileUtils.dissect(FileUtils.java:780)
[19-Jul-2022 02:24:55:142 PDT] 00000021 	at org.apache.tools.ant.util.FileUtils.normalize(FileUtils.java:736)
[19-Jul-2022 02:24:55:142 PDT] 00000021 	at org.apache.tools.ant.util.FileUtils.resolveFile(FileUtils.java:616)
[19-Jul-2022 02:24:55:153 PDT] 00000021 	at org.apache.tools.ant.types.resources.FileResource.<init>(FileResource.java:60)
[19-Jul-2022 02:24:55:153 PDT] 00000021 	at org.apache.tools.ant.types.resources.FileResourceIterator.nextResource(FileResourceIterator.java:143)
[19-Jul-2022 02:24:55:154 PDT] 00000021 	at org.apache.tools.ant.types.resources.FileResourceIterator.next(FileResourceIterator.java:125)
[19-Jul-2022 02:24:55:154 PDT] 00000021 	at org.apache.tools.ant.types.resources.Resources$MyCollection$MyIterator.next(Resources.java:106)
[19-Jul-2022 02:24:55:155 PDT] 00000021 	at org.apache.tools.ant.util.CollectionUtils.asCollection(CollectionUtils.java:219)
[19-Jul-2022 02:24:55:155 PDT] 00000021 	at org.apache.tools.ant.types.resources.Resources$MyCollection.getCache(Resources.java:83)
[19-Jul-2022 02:24:55:156 PDT] 00000021 	at org.apache.tools.ant.types.resources.Resources$MyCollection.iterator(Resources.java:78)
[19-Jul-2022 02:24:55:164 PDT] 00000021 	at org.apache.tools.ant.types.resources.Resources.iterator(Resources.java:170)
[19-Jul-2022 02:24:55:165 PDT] 00000021 	at org.apache.tools.ant.taskdefs.Length.handleResources(Length.java:233)
[19-Jul-2022 02:24:55:165 PDT] 00000021 	at org.apache.tools.ant.taskdefs.Length.eval(Length.java:201)
[19-Jul-2022 02:24:55:165 PDT] 00000021 	at org.apache.tools.ant.taskdefs.condition.And.eval(And.java:42)
[19-Jul-2022 02:24:55:166 PDT] 00000021 	at com.ibm.aries.buildtasks.Iff.execute(Iff.java:122)
[19-Jul-2022 02:24:55:167 PDT] 00000021 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[19-Jul-2022 02:24:55:167 PDT] 00000021 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[19-Jul-2022 02:24:55:168 PDT] 00000021 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[19-Jul-2022 02:24:55:168 PDT] 00000021 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[19-Jul-2022 02:24:55:168 PDT] 00000021 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
[19-Jul-2022 02:24:55:169 PDT] 00000021 	at org.apache.tools.ant.TaskAdapter.execute(TaskAdapter.java:154)
[19-Jul-2022 02:24:55:169 PDT] 00000021 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
[19-Jul-2022 02:24:55:169 PDT] 00000021 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[19-Jul-2022 02:24:55:170 PDT] 00000021 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[19-Jul-2022 02:24:55:170 PDT] 00000021 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[19-Jul-2022 02:24:55:171 PDT] 00000021 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[19-Jul-2022 02:24:55:176 PDT] 00000021 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
[19-Jul-2022 02:24:55:177 PDT] 00000021 	at org.apache.tools.ant.Task.perform(Task.java:348)
[19-Jul-2022 02:24:55:178 PDT] 00000021 	at org.apache.tools.ant.Target.execute(Target.java:392)
[19-Jul-2022 02:24:55:178 PDT] 00000021 	at org.apache.tools.ant.Target.performTasks(Target.java:413)
[19-Jul-2022 02:24:55:179 PDT] 00000021 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
[19-Jul-2022 02:24:55:179 PDT] 00000021 	at org.apache.tools.ant.Project.executeTarget(Project.java:1368)
[19-Jul-2022 02:24:55:179 PDT] 00000021 	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
[19-Jul-2022 02:24:55:180 PDT] 00000021 	at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
[19-Jul-2022 02:24:55:184 PDT] 00000021 	at org.apache.tools.ant.Main.runBuild(Main.java:811)
[19-Jul-2022 02:24:55:185 PDT] 00000021 	at org.apache.tools.ant.Main.startAnt(Main.java:217)
[19-Jul-2022 02:24:55:185 PDT] 00000021 	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
[19-Jul-2022 02:24:55:186 PDT] 00000021 	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
```

Not sure if this is reproducible or not as I don't hit this locally.

--- a/dev/com.ibm.ws.microprofile.graphql_fat_tck/bnd.bnd
+++ b/dev/com.ibm.ws.microprofile.graphql_fat_tck/bnd.bnd
@@ -17,6 +17,9 @@ src: \
 
 fat.project: true
 
+# These features get added programatically
+tested.features: \
+  mpconfig-2.0
 
 # To define a global minimum java level for the FAT, use the following property.
 # If unspecified, the default value is ${javac.source}

--- a/dev/com.ibm.ws.microprofile.graphql_fat_tck/fat/src/org/eclipse/microprofile/graphql/tck/GraphQLTckPackageTest.java
+++ b/dev/com.ibm.ws.microprofile.graphql_fat_tck/fat/src/org/eclipse/microprofile/graphql/tck/GraphQLTckPackageTest.java
@@ -46,10 +46,15 @@ public class GraphQLTckPackageTest {
     @Server("FATServer")
     public static LibertyServer server;
 
+    @BeforeClass
+    public static void setUp() throws Exception {
+        server.startServer();
+    }
+
     @AfterClass
     public static void tearDown() throws Exception {
         if (server != null) {
-            server.postStopServerArchive(); // must explicitly collect since arquillian is starting/stopping the server
+            server.stopServer("CWNEN0047W", "CWNEN0049W", "CWWKZ0014W");
         }
     }
 

--- a/dev/com.ibm.ws.microprofile.rest.client11_fat_tck/bnd.bnd
+++ b/dev/com.ibm.ws.microprofile.rest.client11_fat_tck/bnd.bnd
@@ -17,6 +17,12 @@ src: \
 
 fat.project: true
 
+# These features get added programatically
+tested.features: \
+  servlet-4.0, \
+  cdi-2.0, \
+  jsonp-1.1, \
+  jaxrsclient-2.1
 
 # To define a global minimum java level for the FAT, use the following property.
 # If unspecified, the default value is ${javac.source}

--- a/dev/com.ibm.ws.microprofile.rest.client11_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java
+++ b/dev/com.ibm.ws.microprofile.rest.client11_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java
@@ -11,24 +11,24 @@
 package org.eclipse.microprofile.rest.client.tck;
 
 import org.junit.AfterClass;
+import org.junit.BeforeClass;
 import org.junit.ClassRule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
-import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-import java.util.Map;
-import java.util.HashMap;
-import java.util.HashSet;
+import com.ibm.websphere.simplicity.log.Log;
 
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Map;
 import componenttest.annotation.AllowedFFDC;
 import componenttest.annotation.Server;
 import componenttest.custom.junit.runner.FATRunner;
 import componenttest.rules.repeater.FeatureReplacementAction;
 import componenttest.rules.repeater.RepeatTests;
 import componenttest.topology.impl.LibertyServer;
-import componenttest.topology.impl.JavaInfo;
 import componenttest.topology.utils.MvnUtils;
 
 /**
@@ -39,18 +39,32 @@ import componenttest.topology.utils.MvnUtils;
 public class RestClientTckPackageTest {
 
     @ClassRule
-    public static RepeatTests r = 
+    public static RepeatTests r =
         RepeatTests.withoutModification()
                    .andWith(FeatureReplacementAction.EE8_FEATURES());
 
     @Server("FATServer")
     public static LibertyServer server;
 
+    @BeforeClass
+    public static void setUp() throws Exception {
+    	String javaVersion = System.getProperty("java.version");
+    	Log.info(RestClientTckPackageTest.class, "setup", "javaVersion: " + javaVersion);
+    	System.out.println("java.version = " + javaVersion);
+    	if (javaVersion.startsWith("1.8")) {
+    		Path cwd = Paths.get(".");
+    		Log.info(RestClientTckPackageTest.class, "setup", "cwd = " +  cwd.toAbsolutePath());
+    		Path java8File = Paths.get("publish/tckRunner/tck/tck-suite.xml-java8");
+    	    Path tckSuiteFile = Paths.get("publish/tckRunner/tck/tck-suite.xml");
+    	    Files.copy(java8File, tckSuiteFile, StandardCopyOption.REPLACE_EXISTING);
+    	}
+        server.startServer();
+    }
+
     @AfterClass
     public static void tearDown() throws Exception {
         if (server != null) {
-            server.postStopServerArchive(); // must explicitly collect since arquillian is starting/stopping the server
-//            server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWWKZ0002E");
+            server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWWKZ0002E");
         }
     }
 

--- a/dev/com.ibm.ws.microprofile.rest.client11_fat_tck/publish/servers/FATServer/server.xml
+++ b/dev/com.ibm.ws.microprofile.rest.client11_fat_tck/publish/servers/FATServer/server.xml
@@ -1,9 +1,9 @@
 <!--Copyright (c) 2018 IBM Corporation and others. All rights reserved.
-    This program and the accompanying materials are made available under the 
-    terms of the Eclipse Public License v1.0 which accompanies this distribution, 
-    and is available at 
-        http://www.eclipse.org/legal/epl-v10.html 
-    Contributors: 
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v1.0 which accompanies this distribution,
+    and is available at
+        http://www.eclipse.org/legal/epl-v10.html
+    Contributors:
         IBM Corporation - initial API and implementation
 -->
 <server>
@@ -20,6 +20,6 @@
     <include location="../fatTestPorts.xml" />
 
     <!--Java2 security-->
-    <!-- <javaPermission className="java.security.AllPermission"  name="*" actions="*" /> -->
+    <javaPermission className="java.security.AllPermission"  name="*" actions="*" />
 
 </server>

--- a/dev/com.ibm.ws.microprofile.rest.client12_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java
+++ b/dev/com.ibm.ws.microprofile.rest.client12_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java
@@ -11,21 +11,21 @@
 package org.eclipse.microprofile.rest.client.tck;
 
 import org.junit.AfterClass;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
-import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-import java.util.Map;
-import java.util.HashMap;
-import java.util.HashSet;
+import com.ibm.websphere.simplicity.log.Log;
 
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Map;
 import componenttest.annotation.AllowedFFDC;
 import componenttest.annotation.Server;
 import componenttest.custom.junit.runner.FATRunner;
 import componenttest.topology.impl.LibertyServer;
-import componenttest.topology.impl.JavaInfo;
 import componenttest.topology.utils.MvnUtils;
 
 /**
@@ -38,11 +38,25 @@ public class RestClientTckPackageTest {
     @Server("FATServer")
     public static LibertyServer server;
 
+    @BeforeClass
+    public static void setUp() throws Exception {
+    	String javaVersion = System.getProperty("java.version");
+    	Log.info(RestClientTckPackageTest.class, "setup", "javaVersion: " + javaVersion);
+    	System.out.println("java.version = " + javaVersion);
+    	if (javaVersion.startsWith("1.8")) {
+    		Path cwd = Paths.get(".");
+    		Log.info(RestClientTckPackageTest.class, "setup", "cwd = " +  cwd.toAbsolutePath());
+    		Path java8File = Paths.get("publish/tckRunner/tck/tck-suite.xml-java8");
+    	    Path tckSuiteFile = Paths.get("publish/tckRunner/tck/tck-suite.xml");
+    	    Files.copy(java8File, tckSuiteFile, StandardCopyOption.REPLACE_EXISTING);
+    	}
+        server.startServer();
+    }
+
     @AfterClass
     public static void tearDown() throws Exception {
         if (server != null) {
-            server.postStopServerArchive(); // must explicitly collect since arquillian is starting/stopping the server
-//            server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWWKZ0002E");
+            server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWWKZ0002E");
         }
     }
 

--- a/dev/com.ibm.ws.microprofile.rest.client12_fat_tck/publish/servers/FATServer/server.xml
+++ b/dev/com.ibm.ws.microprofile.rest.client12_fat_tck/publish/servers/FATServer/server.xml
@@ -1,9 +1,9 @@
 <!--Copyright (c) 2019 IBM Corporation and others. All rights reserved.
-    This program and the accompanying materials are made available under the 
-    terms of the Eclipse Public License v1.0 which accompanies this distribution, 
-    and is available at 
-        http://www.eclipse.org/legal/epl-v10.html 
-    Contributors: 
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v1.0 which accompanies this distribution,
+    and is available at
+        http://www.eclipse.org/legal/epl-v10.html
+    Contributors:
         IBM Corporation - initial API and implementation
 -->
 <server>
@@ -18,5 +18,8 @@
     </featureManager>
 
     <include location="../fatTestPorts.xml" />
+
+    <!--Java2 security-->
+    <javaPermission className="java.security.AllPermission"  name="*" actions="*" />
 
 </server>

--- a/dev/com.ibm.ws.microprofile.rest.client13_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java
+++ b/dev/com.ibm.ws.microprofile.rest.client13_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java
@@ -11,21 +11,21 @@
 package org.eclipse.microprofile.rest.client.tck;
 
 import org.junit.AfterClass;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
-import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-import java.util.Map;
-import java.util.HashMap;
-import java.util.HashSet;
+import com.ibm.websphere.simplicity.log.Log;
 
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Map;
 import componenttest.annotation.AllowedFFDC;
 import componenttest.annotation.Server;
 import componenttest.custom.junit.runner.FATRunner;
 import componenttest.topology.impl.LibertyServer;
-import componenttest.topology.impl.JavaInfo;
 import componenttest.topology.utils.MvnUtils;
 
 /**
@@ -38,11 +38,25 @@ public class RestClientTckPackageTest {
     @Server("FATServer")
     public static LibertyServer server;
 
+    @BeforeClass
+    public static void setUp() throws Exception {
+    	String javaVersion = System.getProperty("java.version");
+    	Log.info(RestClientTckPackageTest.class, "setup", "javaVersion: " + javaVersion);
+    	System.out.println("java.version = " + javaVersion);
+    	if (javaVersion.startsWith("1.8")) {
+    		Path cwd = Paths.get(".");
+    		Log.info(RestClientTckPackageTest.class, "setup", "cwd = " +  cwd.toAbsolutePath());
+    		Path java8File = Paths.get("publish/tckRunner/tck/tck-suite.xml-java8");
+    	    Path tckSuiteFile = Paths.get("publish/tckRunner/tck/tck-suite.xml");
+    	    Files.copy(java8File, tckSuiteFile, StandardCopyOption.REPLACE_EXISTING);
+    	}
+        server.startServer();
+    }
+
     @AfterClass
     public static void tearDown() throws Exception {
         if (server != null) {
-            server.postStopServerArchive(); // must explicitly collect since arquillian is starting/stopping the server
-//            server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWWKZ0002E");
+            server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWWKZ0002E", "CWNEN0047W", "CWNEN0049W");
         }
     }
 

--- a/dev/com.ibm.ws.microprofile.rest.client13_fat_tck/publish/servers/FATServer/server.xml
+++ b/dev/com.ibm.ws.microprofile.rest.client13_fat_tck/publish/servers/FATServer/server.xml
@@ -1,9 +1,9 @@
 <!--Copyright (c) 2019 IBM Corporation and others. All rights reserved.
-    This program and the accompanying materials are made available under the 
-    terms of the Eclipse Public License v1.0 which accompanies this distribution, 
-    and is available at 
-        http://www.eclipse.org/legal/epl-v10.html 
-    Contributors: 
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v1.0 which accompanies this distribution,
+    and is available at
+        http://www.eclipse.org/legal/epl-v10.html
+    Contributors:
         IBM Corporation - initial API and implementation
 -->
 <server>
@@ -18,5 +18,8 @@
     </featureManager>
 
     <include location="../fatTestPorts.xml" />
+
+    <!--Java2 security-->
+    <javaPermission className="java.security.AllPermission"  name="*" actions="*" />
 
 </server>

--- a/dev/com.ibm.ws.microprofile.rest.client14_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java
+++ b/dev/com.ibm.ws.microprofile.rest.client14_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java
@@ -11,21 +11,21 @@
 package org.eclipse.microprofile.rest.client.tck;
 
 import org.junit.AfterClass;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
-import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-import java.util.Map;
-import java.util.HashMap;
-import java.util.HashSet;
+import com.ibm.websphere.simplicity.log.Log;
 
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Map;
 import componenttest.annotation.AllowedFFDC;
 import componenttest.annotation.Server;
 import componenttest.custom.junit.runner.FATRunner;
 import componenttest.topology.impl.LibertyServer;
-import componenttest.topology.impl.JavaInfo;
 import componenttest.topology.utils.MvnUtils;
 
 /**
@@ -38,11 +38,25 @@ public class RestClientTckPackageTest {
     @Server("FATServer")
     public static LibertyServer server;
 
+    @BeforeClass
+    public static void setUp() throws Exception {
+    	String javaVersion = System.getProperty("java.version");
+    	Log.info(RestClientTckPackageTest.class, "setup", "javaVersion: " + javaVersion);
+    	System.out.println("java.version = " + javaVersion);
+    	if (javaVersion.startsWith("1.8")) {
+    		Path cwd = Paths.get(".");
+    		Log.info(RestClientTckPackageTest.class, "setup", "cwd = " +  cwd.toAbsolutePath());
+    		Path java8File = Paths.get("publish/tckRunner/tck/tck-suite.xml-java8");
+    	    Path tckSuiteFile = Paths.get("publish/tckRunner/tck/tck-suite.xml");
+    	    Files.copy(java8File, tckSuiteFile, StandardCopyOption.REPLACE_EXISTING);
+    	}
+        server.startServer();
+    }
+
     @AfterClass
     public static void tearDown() throws Exception {
         if (server != null) {
-            server.postStopServerArchive(); // must explicitly collect since arquillian is starting/stopping the server
-//            server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWWKZ0002E");
+            server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWWKZ0002E", "CWNEN0047W", "CWNEN0049W");
         }
     }
 

--- a/dev/com.ibm.ws.microprofile.rest.client14_fat_tck/publish/servers/FATServer/server.xml
+++ b/dev/com.ibm.ws.microprofile.rest.client14_fat_tck/publish/servers/FATServer/server.xml
@@ -1,9 +1,9 @@
 <!--Copyright (c) 2020 IBM Corporation and others. All rights reserved.
-    This program and the accompanying materials are made available under the 
-    terms of the Eclipse Public License v1.0 which accompanies this distribution, 
-    and is available at 
-        http://www.eclipse.org/legal/epl-v10.html 
-    Contributors: 
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v1.0 which accompanies this distribution,
+    and is available at
+        http://www.eclipse.org/legal/epl-v10.html
+    Contributors:
         IBM Corporation - initial API and implementation
 -->
 <server>
@@ -19,5 +19,8 @@
     </featureManager>
 
     <include location="../fatTestPorts.xml" />
+
+    <!--Java2 security-->
+    <javaPermission className="java.security.AllPermission"  name="*" actions="*" />
 
 </server>

--- a/dev/com.ibm.ws.microprofile.rest.client_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java
+++ b/dev/com.ibm.ws.microprofile.rest.client_fat_tck/fat/src/org/eclipse/microprofile/rest/client/tck/RestClientTckPackageTest.java
@@ -10,23 +10,22 @@
  *******************************************************************************/
 package org.eclipse.microprofile.rest.client.tck;
 
-import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
 import java.util.Map;
-import java.util.HashMap;
-import java.util.HashSet;
-
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
+import com.ibm.websphere.simplicity.log.Log;
+
 import componenttest.annotation.AllowedFFDC;
 import componenttest.annotation.Server;
 import componenttest.custom.junit.runner.FATRunner;
 import componenttest.topology.impl.LibertyServer;
-import componenttest.topology.impl.JavaInfo;
 import componenttest.topology.utils.MvnUtils;
 
 /**
@@ -39,16 +38,25 @@ public class RestClientTckPackageTest {
     @Server("FATServer")
     public static LibertyServer server;
 
-//    @BeforeClass
-//    public static void setUp() throws Exception {
-//        server.startServer();
-//    }
-//
+    @BeforeClass
+    public static void setUp() throws Exception {
+    	String javaVersion = System.getProperty("java.version");
+    	Log.info(RestClientTckPackageTest.class, "setup", "javaVersion: " + javaVersion);
+    	System.out.println("java.version = " + javaVersion);
+    	if (javaVersion.startsWith("1.8")) {
+    		Path cwd = Paths.get(".");
+    		Log.info(RestClientTckPackageTest.class, "setup", "cwd = " +  cwd.toAbsolutePath());
+    		Path java8File = Paths.get("publish/tckRunner/tck/tck-suite.xml-java8");
+    	    Path tckSuiteFile = Paths.get("publish/tckRunner/tck/tck-suite.xml");
+    	    Files.copy(java8File, tckSuiteFile, StandardCopyOption.REPLACE_EXISTING);
+    	}
+        server.startServer();
+    }
+
     @AfterClass
     public static void tearDown() throws Exception {
         if (server != null) {
-            server.postStopServerArchive(); // must explicitly collect since arquillian is starting/stopping the server
-//            server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWWKZ0002E");
+            server.stopServer("CWMCG0007E", "CWMCG0014E", "CWMCG0015E", "CWMCG5003E", "CWWKZ0002E");
         }
     }
 

